File: (383)Ransom Note.py
from collections import Counter


class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        lookup = Counter(magazine)
        #print(lookup)
        for c in ransomNote:
            if lookup[c] == 0:
                return False
            lookup[c] -= 1
        return True

File: (383)Ransom Note.cpp
class Solution {
    vector<int> getFreq(string& s) {
        vector<int> freq(26, 0);
        for (auto& letter: s) 
            freq[letter - 'a']++;
        return freq;
    }
public:
    bool canConstruct(string ransomNote, string magazine) {
        auto freqRansom = getFreq(ransomNote), freqMag = getFreq(magazine);
        for (int i = 0; i < 26; ++i) {
            if (freqRansom[i] > freqMag[i]) {
                return false;
            }
        }
        return true;
    }
};